packages/@okta/vuepress-site/docs/guides/build-sso-integration/main/openidconnect/prep.md

```diff
@@ -1,33 +1,150 @@
-1. Based on the [type of application that you have built](/docs/concepts/oauth-openid/#what-kind-of-client-are-you-building), determine the correct [OAuth 2.0 flow](/docs/concepts/oauth-openid/#recommended-flow-by-application-type) that is required below the OIDC identity layer.
+For concurrent rate limits, the three headers behave a little differently:
 
-    For OIDC applications destined for the OIN, you can create either of the following:
+* When the number of unfinished requests is below the concurrent rate limit, request headers only report org-wide rate limits.
+* After you exceed a concurrent rate limit, the headers report that the limit has been exceeded.
+* When you drop back down below the concurrent rate limit, the headers switch back to reporting the time-based rate limits.
+* The first two header values are always `0` for concurrent rate limit errors. The third header reports an estimated time interval when the concurrent rate limit may be resolved.
+* The `X-Rate-Limit-Reset` time for concurrent rate limits is only a suggested value. There's no guarantee that enough requests can complete for the requests to go below the concurrent rate limit at the time indicated.
 
-    * A Web application with a dedicated server-side back-end that is capable of securely storing a Client Secret and exchanging information with an authorization server through trusted back-channel connections. Okta recommends using the Authorization Code flow with an exchange of the client credentials (Client ID and Client Secret) for controlling the access between your application and the resource server.
-    * A Single Page Application (SPA) that uses an Authorization Code flow with a Proof Key for Code Exchange (PKCE). Okta recommends this method to control the access between your SPA application and a resource server.
+The error condition resolves itself when there's another concurrent thread available. Normally no intervention is required. You may be exceeding the concurrent rate limit if you notice frequent bursts of HTTP 429 errors. Examine the activities in the log before the burst of HTTP 429 errors appeared. If you can't identify what is causing you to exceed the limit, contact [Okta Support](mailto:support@okta.com).
 
-1. Determine the sign-in redirect URIs on your system. A redirect URI is where Okta sends the authentication response and ID token during the sign-in flow. You can specify more than one URI if required.
-1. Your application must support automatic credential rotation. For more information, see the `/keys` section in the [OpenID Connect & OAuth 2.0 API reference](/docs/reference/api/oidc/#key-rotation).
+You can request a temporary rate limit increase if you anticipate a large number of requests over a specified time period. Contact [Okta Support](mailto:support@okta.com) to open a ticket to permit the exception.
```

Comment:
might be helpful to link to https://support.okta.com/help/s/article/How-can-we-request-to-have-the-rate-limit-for-our-org-temporarily-increased?language=en_US if its simple/kosher to link to support.okta from developer.okta